akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -1359,6 +1358,27 @@ final class Flow[In, Out, Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends Gr
   def recoverWith(pf: PartialFunction[Throwable, _ <: Graph[SourceShape[Out], NotUsed]]): javadsl.Flow[In, Out, Mat] =
     new Flow(delegate.recoverWith(pf))
 
+  /**
+   * RecoverWithComplete allows to transform a stream failure into a successful stream.
+   *
+   * Since the underlying failure signal onError arrives out-of-band, it might jump over existing elements.
+   * This stage can recover the failure signal, but not the skipped elements, which will be dropped.
+   *
+   * Throwing an exception inside `recoverTermination` _will_ be logged on ERROR level automatically.
+   *
+   * '''Emits when''' element is available from the upstream
+   *
+   * '''Backpressures when''' downstream backpressures
+   *
+   * '''Completes when''' upstream completes or upstream failed
+   *
+   * '''Cancels when''' downstream cancels
+   */
+  def recoverWithComplete(ex: Class[Throwable]): javadsl.Flow[In, Out, Mat] =
+    new Flow(delegate.recoverWithComplete {
+      case t if ex == t.getClass ⇒ Done
+    })
```

Comment:
since we started this a new `def recoverWith(clazz: Class` was added. It's using instanceof, which this one should also use.

```
case elem if clazz.isInstance(elem) =>
```

Another difference is

```
clazz: Class[_ <: Throwable]
```

Reply:
@patriknw sure, will have a look! could you explain the difference of `Class[Throwable]` and `Class[_ <: Throwable]` briefly?